packages/remix-server-runtime/router/utils.ts

```diff
@@ -114,6 +114,24 @@ export interface ActionFunction {
   (args: ActionFunctionArgs): Promise<Response> | Response | Promise<any> | any;
 }
 
+/**
+ * Route loader function signature.
+ *
+ * This is intended to be used with TypeScript's `satisfies` keyword.
+ *
+ * @example
+ *
+ * ```ts
+ * export const loader = (({ request }) => {
+ *   const url = new URL(request.url);
+ *   return new Response("hello from " + url.pathname);
+ * }) satisfies LoaderFunction;
+ * ```
+ */
+export interface LoaderFunction {
+  (args: LoaderFunctionArgs): Promise<Response> | Response;
```

Comment:
We may need to iterate on the return type a little.

Eg. do we want to allow loader functions to return `null | Response`?